odoo/addons/base/models/ir_attachment.py

```diff
@@ -532,7 +532,7 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
         discard_binary_fields_attachments = False
         if not any(arg[0] in ('id', 'res_field') for arg in args):
             discard_binary_fields_attachments = True
-            args.insert(0, ('res_field', '=', False))
+            args.insert(0, ('res_field', 'in', [False, 'hidden']))
```

Comment:
Attachments with a specified `res_field` don't appear in the attachments view. So to still be able to see the images that have been removed from the media dialog but not from the website (and so not from the DB), this is the simplest and least intrusive way I found.

Reply:
Maybe worth considering to add a new field on the attachment model then. In the Website override of course so that we don't touch `base`. Or is there any issue with that ?